jupyter-pytorch-full/tests/test_access.py

```diff
@@ -8,9 +8,20 @@
 import os
 import subprocess
 import time
+import requests
+import tenacity
 import yaml
 
 
+@tenacity.retry(
+stop=tenacity.stop_after_attempt(5),
+wait=tenacity.wait_fixed(2)
+)
+def check_notebook_server_up(url):
+    response = requests.get(url)
+    response.raise_for_status() # Raise an exception if the request was unsuccessful
+    return response.text
```

Comment:
do you really need to return something ?

Reply:
I would still return some result. Or even better return `True` if response status is 200 and `False` otherwise.

Reply:
we need to return the response because then we are asserting on it:
```
assert "JupyterLab" in output
```